programming_examples/channel_examples/worker_to_self/worker_to_self.py

```diff
@@ -47,51 +51,43 @@ def launch_body(a, b):
             @segment(name="seg")
             def segment_body():
 
+                tensor_in_l2 = AllocOp(image_type_l2, [], [])
+                ChannelGet("ChanIn", tensor_in_l2)
+
                 # The herd sizes correspond to the dimensions of the contiguous block of cores we are hoping to get.
                 # We just need one compute core, so we ask for a 1x1 herd
-                @herd(name="copyherd", sizes=[1, 1])
-                def herd_body(tx, ty, sx, sy):
+                @herd(
+                    name="copyherd",
+                    sizes=[1, 1],
+                    operands=[tensor_in_l2],
+                )
+                def herd_body(tx, ty, sx, sy, tensor_in_l2):
 
                     # We must allocate a buffer of image size for the input/output
-                    tensor_in = AllocOp(image_type, [], [])
-                    tensor_out = AllocOp(image_type, [], [])
-                    tensor_in2 = AllocOp(image_type, [], [])
-                    tensor_out2 = AllocOp(image_type, [], [])
+                    tensor_in_l1 = AllocOp(image_type_l1, [], [])
+                    tensor_out_l1 = AllocOp(image_type_l1, [], [])
 
-                    ChannelGet("ChanIn", tensor_in)
+                    ChannelPut("ToSelf", tensor_in_l2)
```

Comment:
I see. I'd sugges moving this channel put to outside herd.

Reply:
This example is called worker to self because the purpose of the example is to demonstrate (if) it is possible to put data into a channel and then fetch it from the same channel from within the same worker/compute core in a herd. So to keep this example, we need both `ChannelPut("ToSelf", some_mem)` and `ChannelGet("ToSelf", some_other_mem)` within the herd. 

The only reason I modified this example in this PR is because previously this example was blocked because I was unable to access L2 memory in the herd, and that capability is needed because channels must move data between different memory regions (in this case, L2 and L1).

Reply:
I see. This scenario is not exercised before.